src/gene/cli.py

```diff
@@ -2,67 +2,140 @@
+@click.option("--normalize", is_flag=True, help="Create normalized records.")
+@click.option("--db_url", help=url_description)
+@click.option("--aws_instance", is_flag=True, help="Use cloud DynamodDB instance.")
+@click.option(
+    "--use_existing",
+    is_flag=True,
+    default=False,
+    help="Use most recent locally-available source data instead of fetching latest version",
+)
+@click.option(
+    "--silent", "-s", is_flag=True, default=False, help="Suppress console output."
+)
+def update(
+    sources: Tuple[str],
+    aws_instance: bool,
+    db_url: str,
+    all: bool,
+    normalize: bool,
+    use_existing: bool,
+    silent: bool,
+) -> None:
+    """Update provided normalizer SOURCES in the gene database.
+
+    Valid SOURCES are "HGNC", "NCBI", and "Ensembl" (case is irrelevant). SOURCES are
+    optional, but if not provided, either --all or --normalize must be used.
+
+    For example, the following command will update NCBI and HGNC source records:
+
+        $ gene-normalizer update HGNC NCBI
+
+    To completely reload all source records and construct normalized concepts, use the
+    --all and --normalize options:
 
-@click.command()
+        $ gene-normalizer update --all --normalize
+
+    The Gene Normalizer will fetch the latest available data from all sources if local
+    data is out-of-date. To suppress this and force usage of local files only, use the
+    --use_existing flag:
+
+        $ gene-normalizer update --all --use_existing
+
+    \f
+    :param sources: tuple of raw names of sources to update
+    :param aws_instance: if true, use cloud instance
+    :param db_url: URI pointing to database
+    :param all: if True, update all sources (ignore ``sources``)
+    :param normalize: if True, update normalized records
+    :param use_existing: if True, use most recent local data instead of fetching latest version
+    :param silent: if True, suppress console output
+    """  # noqa: D301
+    if (not sources) and (not all) and (not normalize):
+        click.echo(
+            "Error: must provide SOURCES or at least one of --all, --normalize\n"
+        )
+        ctx = click.get_current_context()
+        click.echo(ctx.get_help())
+        ctx.exit(1)
+
+    db = create_db(db_url, aws_instance, silent)
+
+    processed_ids = None
+    if all:
+        processed_ids = update_all_sources(db, use_existing, silent=silent)
+    elif sources:
+        parsed_sources = []
+        failed_source_names = []
+        for source in sources:
+            try:
+                parsed_sources.append(SourceName[source.upper()])
+            except KeyError:
+                failed_source_names.append(source)
+        if len(failed_source_names) != 0:
+            click.echo(f"Error: unrecognized sources: {failed_source_names}")
+            click.echo(f"Valid source options are {list(SourceName)}")
+            click.get_current_context().exit(1)
+
+        parsed_sources = list(set(parsed_sources))
```

Comment:
I don't think it's necessary to cast back to list. We could just have `parsed_sources` be a set in L93

```suggestion
        parsed_sources = set(parsed_sources)
```